app/models/user.rb

```diff
@@ -9,12 +9,33 @@ class User < ApplicationRecord
   validate :validate_image_type
 
   has_one_attached :profile_picture
-  has_many :posts
+  has_many :posts # , dependent:   :destroy ??
+
+  has_many :active_relationships, class_name: 'FollowRelationship',
+                                  foreign_key: 'follower_id',
+                                  dependent: :destroy
+  has_many :passive_relationships, class_name: 'FollowRelationship',
+                                   foreign_key: 'followed_id',
+                                   dependent: :destroy
+  has_many :following, through: :active_relationships, source: :followed
+  has_many :followers, through: :passive_relationships, source: :follower
 
   def full_name
     "#{first_name} #{last_name}"
   end
 
+  def follow(other_user)
+    active_relationships.create(followed_id: other_user.id)
+  end
+
+  def unfollow(other_user)
+    active_relationships.find_by(followed_id: other_user.id).destroy
```

Comment:
What if the js failed to reload and the user hit the button twice? You'll need to add safe navigation in case the find_by returns nil!